internal/controllers/nodequotaconfig_controller.go

```diff
@@ -68,47 +77,56 @@ func (r *NodeQuotaConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&danav1alpha1.NodeQuotaConfig{}).
 		Watches(
-			&source.Kind{Type: &corev1.Node{}},
+			&corev1.Node{},
 			handler.EnqueueRequestsFromMapFunc(r.requestConfigReconcile),
 		).
 		Complete(r)
 }
 
-func (r *NodeQuotaConfigReconciler) CalculateRootSubnamespaces(ctx context.Context, config danav1alpha1.NodeQuotaConfig) error {
-	oldConfigStatus := config.Status.DeepCopy()
-	reservedResources := []danav1alpha1.ReservedResources{}
+// CalculateRootSubnamespaces calculates the resource allocation for the root subnamespaces based on the provided NodeQuotaConfig.
+// It takes a context, the NodeQuotaConfig to reconcile, and a logger for logging informational messages.
+// It returns an error (if any occurred) during the calculation.
+func (r *NodeQuotaConfigReconciler) CalculateRootSubnamespaces(ctx context.Context, config danav1alpha1.NodeQuotaConfig, logger logr.Logger) error {
 	for _, rootSubnamespace := range config.Spec.Roots {
+		logger.Info(fmt.Sprintf("Starting to calculate RootSubnamespace %s", rootSubnamespace.RootNamespace))
```

Comment:
I think it would be better in terms of readability to move the `for` loop into the main Reconcile code and pass the rootNamespace name as an argument to this function. WDYT?

Also, the root namespace is not a subnamespace (there's no subnamespace CR for it)